bencode-parser-fuzz/bencode-parser-fuzz/src/main.rs

```diff
@@ -0,0 +1,86 @@
+#[macro_use]
+extern crate honggfuzz;
+use nom_bencode::Value;
+use bencoder::bencode::{Bencode, BencodeError};
+use std::collections::{BTreeMap, HashMap};
+
+
+fn main() {
+    println!("Starting fuzzer");
+    loop {
+        fuzz!(|data: &[u8]| {
+            let nom_result = parse_source_nom(data).unwrap();
+            let libtorrent_result = parse_source_libtorrent(data).unwrap();
+            
+            assert_eq!(to_bencode(nom_result), libtorrent_result);
+        });
+    }
+}
+
+fn parse_source_nom(src: &[u8]) -> Result<Vec<nom_bencode::Value>, nom_bencode::Error<&[u8]>> {
+    nom_bencode::parse(src)
+}
+
+fn parse_source_libtorrent(src: &[u8]) -> Result<Bencode, BencodeError> {
+    Bencode::decode(&src.to_vec())
+}
+
+fn to_bencode(value: Vec<Value>) -> Bencode {
+    
+    match &value[..] {
+        [Value::Integer(value), _] => Bencode::BNumber(*value),
+        [Value::Bytes(value),_] => to_string_bencode(value),
+        [Value::List(value), _] => to_list_bencode(value),
+        [Value::Dictionary(value), _] => to_dict_bencode(value),
+        [] | [_] | [_, _, _, ..] => to_bencode(value),
```

Comment:
You're passing `value` unmodified recursively to `to_bencode`, so I think this triggers infinite recursion. I don't entirely understand the last case in this line either. This should just be the terminating case I believe.

Reply:
fixed! now it panic in that case, it has to be one of the previous values or else it makes no sense